include/ESA.hpp

```diff
@@ -0,0 +1,37 @@
+#ifndef ESA_HPP
+#define ESA_HPP
+
+#include <limits>
```

Comment:
If these include statements are not necessary, remove them. To check if they are necessary just try to remove them and recompile, if the code compiles then they are not necessary